src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -51,26 +52,49 @@ public AddCommand parse(String args) throws ParseException {
         Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
         Optional<String> medicalAge = argMultimap.getValue(PREFIX_AGE);
         Optional<String> medicalString = argMultimap.getValue(PREFIX_MEDICAL);
+        Optional<String> time = argMultimap.getValue(PREFIX_SCHEDULE);
+        Person person;
+        LocalDateTime parsedTime = null;
+        if (time.isPresent()) {
+            if (time.get().trim().length() > 0) {
+                parsedTime = ParserUtil.parseTime(time.get());
+            }
+        }
         if (medicalString.isEmpty()) {
             if (medicalAge.isEmpty()) {
-                Person person = new Person(name, phone, email, address, tagList);
-                return new AddCommand(person);
+                if (time.isEmpty()) {
+                    person = new Person(name, phone, email, address, tagList);
+                } else {
+                    person = new Person(name, phone, email, address, tagList, parsedTime);
+                }
             } else {
                 Age age = ParserUtil.parseAge(medicalAge.get().toString());
-                Person person = new Person(name, phone, email, address, age, tagList);
-                return new AddCommand(person);
+                if (time.isEmpty()) {
+                    person = new Person(name, phone, email, address, age, tagList);
+                } else {
+                    person = new Person(name, phone, email, address, age, tagList, parsedTime);
+                }
             }
         } else {
             MedicalCondition medicalCondition = ParserUtil.parseMedicalCond(medicalString.get());
             if (medicalAge.isEmpty()) {
-                Person person = new Person(name, phone, email, address, tagList, medicalCondition);
-                return new AddCommand(person);
+                if (time.isEmpty()) {
+                    person = new Person(name, phone, email, address, tagList, medicalCondition);
+                } else {
+                    person = new Person(name, phone, email, address, tagList,
+                        parsedTime, medicalCondition);
+                }
             } else {
                 Age age = ParserUtil.parseAge(medicalAge.get().toString());
-                Person person = new Person(name, phone, email, address, age, tagList, medicalCondition);
-                return new AddCommand(person);
+                if (time.isEmpty()) {
+                    person = new Person(name, phone, email, address, age, tagList, medicalCondition);
+                } else {
+                    person = new Person(name, phone, email, address, age, tagList,
+                        parsedTime, medicalCondition);
+                }
             }
         }
```

Comment:
So far this part quite nested. SLAP not satisfied. But it works for functionality.